src/Symfony/Component/Serializer/Mapping/ClassDiscriminatorResolver.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Mapping;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+final class ClassDiscriminatorResolver implements ClassDiscriminatorResolverInterface
+{
+    /**
+     * @var ClassDiscriminatorMapping[]
+     */
+    private $mapping = array();
+
+    /**
+     * @param string                    $class
+     * @param ClassDiscriminatorMapping $mapping
+     */
+    public function addClassMapping($class, ClassDiscriminatorMapping $mapping)
+    {
+        if (isset($this->mapping[$class])) {
```

Comment:
How about injecting the mappings through a constructor and making this verification at container compile time, not runtime?